src/keybaseca/bot/bot.go

```diff
@@ -131,25 +142,143 @@ func StartBot(conf config.Config) error {
 	}
 }
 
-// Log the given error to Keybase chat and to the configured log file. Used so that the chatbot does not crash
-// due to an error caused by a malformed message.
-func LogError(conf config.Config, kbc *kbchat.API, msg kbchat.SubscriptionMessage, err error) {
+// Write kssh config for kssh to use
+func (b *Bot) writeClientConfig() error {
+	username := b.api.GetUsername()
+	if username == "" {
+		return fmt.Errorf("failed to get a username from kbChat, got an empty string")
+	}
+
+	teams := b.conf.GetTeams()
+	if b.conf.GetChatTeam() != "" {
+		// Make sure we write the kssh config in the chat team, which may not be in
+		// the list of teams
+		teams = append(teams, b.conf.GetChatTeam())
+	}
+
+	// If they configured a chat team, have messages go there
+	config := kssh.Config{TeamName: b.conf.GetChatTeam(), BotName: username, ChannelName: b.conf.GetChannelName()}
+
+	for _, team := range teams {
+		if b.conf.GetChatTeam() == "" {
+			// If they didn't configure a chat team, messages should be sent to any
+			// channel. This is done by having each client config reference the team
+			// it is found in.
+			config.TeamName = team
+			config.ChannelName = ""
+		}
+
+		var bytes []byte
+		bytes, err := json.Marshal(config)
+		if err != nil {
```

Comment:
inside this error is a pretty good place to drop a log for which team had an issue